custom_components/ha-glocaltokens/api.py

```diff
@@ -46,3 +57,69 @@ def get_google_devices_json(self):
     def get_android_id(self):
         """Generate random android_id"""
         return self._client._get_android_id()
+
+    def format_timer_information(timer):
+        timestamp_ms = timer[FIRE_TIME]
+        timestamp = timestamp_ms / 1000
+        humantime = datetime.fromtimestamp(timestamp).strftime(SHOW_TIME_ONLY)
+        timer[DATE_TIME] = humantime
+
+        duration_ms = timer[ORIGINAL_DURATION]
+        duration = duration_ms / 1000
+        humanduration = datetime.utcfromtimestamp(duration).strftime(SHOW_TIME_ONLY)
+        timer[DURATION] = humanduration
+        return timer
+
+    def format_alarm_information(alarm):
+        timestamp_ms = alarm[FIRE_TIME]
+        timestamp = timestamp_ms / 1000
+        humantime = datetime.utcfromtimestamp(timestamp).strftime(SHOW_DATE_TIMEZONE)
+        localtime = datetime.fromtimestamp(timestamp).strftime(SHOW_DATE_AND_TIME)
+        alarm[DATE_TIME] = humantime
+        alarm[LOCAL_TIME] = localtime
+        return alarm
+
+    def create_url(ip):
+        url = 'https://'+ ip +':'+ str(PORT) + ENDPOINT
+        return url
+
+    def get_alarms_and_timers_from(ip, token):
+
+        url = GlocaltokensApiClient.create_url(ip)
+
+        HEADERS[HEADER_CAST_LOCAL_AUTH] = token
+
+        response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT)
+
+        return response
+
+    def get_google_devices_information(self):
+
+        devices = self._client.get_google_devices_json()
+
+        for x in range(len(devices)):
+            #local_token = next(item[GLOCALTOKENS_TOKEN] for item in devices)
+            local_token = devices[1][GLOCALTOKENS_TOKEN]
```

Comment:
Then here you can just say `local_token = device[GLOCALTOKENS_TOKEN]`

Reply:
but devices from get_google_devices_json() is a list not a dict

Reply:
See above

Reply:
This seems still valid, so i'll unresolve it